src/main/java/com/google/dicomwebfuse/entities/cache/Cache.java

```diff
@@ -210,25 +210,33 @@ public void setInstanceCommand(DicomPath dicomPath, Command command) throws Dico
   }
 
   public boolean isDatasetOutdated() {
-    return cachedDataset.getDatasetCacheTime().isBefore(Instant.now());
+    Instant instantNow = Instant.now();
+    return cachedDataset.getDatasetCacheTime().isBefore(instantNow) ||
+        cachedDataset.getDatasetCacheTime().equals(instantNow);
```

Comment:
Why did you have to make this change?

Reply:
Should there be unit tests covering this?

Reply:
> Why did you have to make this change?

Instant.now()  - Returns a value in microseconds in Java 9 and later, but is restricted to mere milliseconds in Java 8. That's why in some cases during a unit test(DicomFuseHelperTest) I got heisenbug. Sometimes tests were failed sometimes were successful. I found that code may return false if instantNow is less than 1 millisecond(if code faster than 1 millisecond).

> Should there be unit tests covering this?

This was done in the DicomFuseHelperTest. If the tests are successful it means that this code works.

Reply:
Ok please add a code comment explaining this.

Reply:
Done